base/process.jl

```diff
@@ -679,7 +679,9 @@ process_signaled(s::Process) = (s.termsignal > 0)
 function process_status(s::Process)
     return process_running(s) ? "ProcessRunning" :
            process_signaled(s) ? "ProcessSignaled(" * string(s.termsignal) * ")" :
-           process_exited(s) ? "ProcessExited(" * string(s.exitcode) * ")" :
+           process_exited(s) ? "ProcessExited(" * (Sys.iswindows() ? ("0x" * uppercase(string(s.exitcode, 16))) : string(s.exitcode)) * (
```

Comment:
`string(s.exitcode, 16)`, I don't think this does what you want:

```julia
julia> s = run(`echo foo`);
foo

julia> string(s.exitcode, 16)
"016"

julia> s.exitcode
0
```

Hence why a test is good.